third_party/mbedtls/mbedtls-config.h

```diff
@@ -33,6 +33,7 @@
 
 #include "openthread-core-config.h"
 
+#include <limits.h>
 #include <stdio.h>
 #include <stdlib.h>
```

Comment:
I don't think any of these are necessary, but they're harmless.

Reply:
Removing `limits.h` results in the following build error:

```
/src/openthread/third_party/mbedtls/repo/library/oid.c:775:16: error: use of undeclared identifier 'INT_MAX'
  775 |     if (size > INT_MAX) {
      |                ^
/src/openthread/third_party/mbedtls/repo/library/oid.c:787:22: error: use of undeclared identifier 'UINT_MAX'
  787 |         if (value > (UINT_MAX >> 7)) {
      |                      ^
2 errors generated.
```

Example: https://github.com/openthread/openthread/actions/runs/9131072614/job/25109469015

Thoughts?

Reply:
Hmmm, this is a bug in Mbed TLS. `oid.c` uses a symbol from `limits.h`, so going by our coding rules, it should include `limits.h`. Obviously this can go undetected since `limit.h` is a pretty common header and happens to always be included from some header in all the builds we do.

However, there seems to be something else going on, because `limits.h` is included from `mbedtls/build_info.h` which is included from `mbedtls/oid.h` which is included from `oid.c`. What version of Mbed TLS is in that build?

Reply:
I believe it is using 2.28.7 and specifically https://github.com/Mbed-TLS/mbedtls/commit/555f84735a.

Reply:
Oh, I see. It's an actual bug in Mbed TLS 2.28, and just a misdemeanor in Mbed TLS ≥3.0. I'll push a fix on our end, but in the meantime including `limits.h` here is a harmless workaround.

If I understand correctly, this configuration is meant to be compatible both with Mbed TLS 2.28 and 3.x? If so you can still include `check_config.h` (which includes `limits.h` in 2.28) in Mbed TLS 2.x only:

```
#include "mbedtls/version.h"
#if (MBEDTLS_VERSION_NUMBER < 0x03000000)
    // Configuration sanity check. Done automatically in Mbed TLS >= 3.0.
    #include "mbedtls/check_config.h"
#endif
```

Reply:
Yes, intent is for it to be compatible with both 2.28 and 3.x, at least for now. I just pushed a change with your suggestion. Thanks!

Reply:
Removing `stdlib.h` results in this build errors :)

```
Users/runner/work/openthread/openthread/third_party/mbedtls/repo/library/memory_buffer_alloc.c:246:9: error: implicitly declaring library function 'exit' with type 'void (int) __attribute__((noreturn))' [-Werror,-Wimplicit-function-declaration]
        mbedtls_exit(1);
        ^
/Users/runner/work/openthread/openthread/third_party/mbedtls/repo/include/mbedtls/platform.h:322:24: note: expanded from macro 'mbedtls_exit'
#define mbedtls_exit   exit
                       ^
/Users/runner/work/openthread/openthread/third_party/mbedtls/repo/library/memory_buffer_alloc.c:246:9: note: include the header <stdlib.h> or explicitly provide a declaration for 'exit'
/Users/runner/work/openthread/openthread/third_party/mbedtls/repo/include/mbedtls/platform.h:322:24: note: expanded from macro 'mbedtls_exit'
#define mbedtls_exit   exit
                       ^
1 error generated.
```

https://github.com/openthread/openthread/actions/runs/9131493121/job/25110806918?pr=10263

Reply:
I see. That's got to be [a bug in Mbed TLS](https://github.com/Mbed-TLS/mbedtls/issues/9154) as well. But it's much more subtle to figure out and fix (without breaking the build on freestanding platforms that don't have `stdio.h` or `stdlib.h`), so I can't promise we'll manage to fix it any time soon.